src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -0,0 +1,63 @@
+"""
+Thread used to monitor the agent draining process. Should
+eventually report no issues meaning that the agent is ready
+to be shutdown and a new version be put in place.
+"""
+from __future__ import division
+
+__all__ = []
+
+import logging
+import traceback
+from WMCore.WorkerThreads.BaseWorkerThread import BaseWorkerThread
+from WMComponent.AnalyticsDataCollector.DataCollectAPI import isDrainMode
+
+
+class DrainStatusPoller(BaseWorkerThread):
+    """
+    Collects information related to the agent drain status
+    """
+    # class variable that contains drain statistics
+    drainStats = {}
+
+    def __init__(self, config):
+        """
+        initialize properties specified from config
+        """
+        BaseWorkerThread.__init__(self)
+        self.config = config
+
+    def algorithm(self, parameters):
+        """
+        Update drainStats if agent is in drain mode
+        """
+        if isDrainMode(self.config):
+            logging.info("Checking agent drain status...")
+
+            try:
+                DrainStatusPoller.drainStats = self.collectDrainInfo()
+                logging.info("Finished collecting agent drain status.")
+
+            except Exception as ex:
+                msg = "Error occurred, will retry later:\n"
+                msg += str(ex)
+                msg += "Trace back: \n%s" % traceback.format_exc()
```

Comment:
this traceback line can be removed (and the import). the logging.exception line below will automatically print the traceback.